ivadomed/main.py

```diff
@@ -656,6 +665,35 @@ def cmd_test(context):
             else:
                 preds = model(test_input)
 
+        # WARNING: sample['gt'] is actually the pred in the return sample
+        # implementation justification: the other option: rdict['pred'] = preds would require to largely modify mt_transforms
+        rdict = {}
+        rdict['gt'] = preds.cpu()
+        batch.update(rdict)
+
+        # reconstruct 3D image
+        for smp_idx in range(len(batch['gt'])):
+            # undo transformations
+            rdict = {}
+            for k in batch.keys():
+                rdict[k] = batch[k][smp_idx]
+            rdict_undo = val_undo_transform(rdict)
+
+            fname_ref = rdict_undo['input_metadata']['gt_filename']
+            if pred_tmp_lst and (fname_ref != fname_tmp or (i == len(test_loader)-1 and smp_idx == len(batch['gt'])-1)):  # new processed file
+                # save the completely processed file as a nii
+                fname_pred = os.path.join(path_3Dpred, fname_tmp.split('/')[-1])
+                fname_pred = fname_pred.split('manual.nii.gz')[0] + 'pred.nii.gz'
+                save_nii(pred_tmp_lst, z_tmp_lst, fname_tmp, fname_pred, axis_dct[context['slice_axis']])
+                # re-init pred_stack_lst
+                pred_tmp_lst, z_tmp_lst = [], []
+                # TODO compute image-based metrics
+
+            # add new sample to pred_tmp_lst
+            pred_tmp_lst.append(np.array(rdict_undo['input']))
```

Comment:
pred_tmp_lst should contain the prediction, should it be rdict_undo['gt'] instead of rdict_undo['input']?

Reply:
ahah yes absolutely! I used input for my testing, but yes of course gt :)